src/CLI.hs

```diff
@@ -0,0 +1,215 @@
+{-# LANGUAGE OverloadedStrings #-}
+module CLI (
+  LogLevel(..),
+  Component(..),
+  Command(..),
+  Options(..),
+
+  run
+) where
+
+import Data.Text hiding (empty)
+import Options.Applicative
+
+--------------------------------------------------------------------------------
+                             -- Types --
+--------------------------------------------------------------------------------
+data LogLevel
```

Comment:
We already have a datatype for those, the constructors aren't exported but it has Bounded and Enum instances which should be enough for what you need to translate the numbers.